Source/ViewControllers/BrickCollectionView.swift

```diff
@@ -531,14 +539,10 @@ extension BrickCollectionView: UICollectionViewDataSource {
 
         if let brickCell = cell as? BrickCell {
             if var resizable = brickCell as? AsynchronousResizableCell {
-                resizable.sizeChangedHandler = { [weak brickCollectionView] resizedCell in
-                    guard let brickCollectionView = brickCollectionView else {
-                        return
-                    }
-                    let height = resizedCell.heightForBrickView(withWidth: resizedCell.frame.width)
-                    brickCollectionView.performBatchUpdates({
-                        brickCollectionView.layout.updateHeight(indexPath, newHeight: height)
-                        }, completion: nil)
+                resizable.sizeChangedHandler = { [weak self] in
+                    // A strong reference to the cell forces it (brickCell) to remain in memory for the duration of the batch update.
+                    // The downside, is that we must always remember to call the sizeChangedHandler in order for the cell to be released from memory.
+                    self?.performResize(brickCell, indexPath: indexPath)
```

Comment:
`brickCell` appears to be strongly referenced in this closure which would reintroduce the memory leak. How would calling `sizeChangedHandler` break this cycle?